internal/controlplane/handlers_oauth.go

```diff
@@ -162,6 +175,26 @@ func (s *Server) HandleProviderCallback() runtime.HandlerFunc {
 	}
 }
 
+// HandleGitHubAppCallback handles the authorization callback from the GitHub App. This function validates the
+// GitHub user has access to the installation. It also gathers the state from the database and compares it to
+// the state passed in, if present. If they match a new GitHub App provider is created with the installation ID.
+// note: this is an HTTP only (not RPC) handler
+func (s *Server) HandleGitHubAppCallback() runtime.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request, pathParams map[string]string) {
+		ctx := r.Context()
+
+		if err := s.processAppCallback(ctx, w, r, pathParams); err != nil {
+			if httpErr, ok := err.(*httpResponseError); ok {
```

Comment:
should this be an `errors.Is` instead in case `processAppCallback` returns a wrapped error (I know we have the same issue in `HandleProviderCallback`

Reply:
I think you mean `errors.As`, since we'd want to extract the wrapped error as well in that case in order to be able to call `WriteResponse`.